Add ckeck for Direct3D

egomeh opened the pull request:
The raymarching shader now has a uniform that sets the sign of the y axis of uv of the ray being cast in the pixel shader
In case if Direct3D the rays were directed upside down

egomeh committed:
Add ckeck for Direct3D
The raymarching shader now has a uniform that sets the sign of the y axis of uv of the ray being cast in the pixel shader
In case if Direct3D the rays were directed upside down

gokselgoktas commented on Assets/Resources/Shaders/RayMarching.shader:
```diff
@@ -43,8 +45,12 @@ Shader "Hidden/Ray Marching"
         Varyings output;
 
+        // Handles vertically-flipped case.
+        float vflip = sign(_MainTex_TexelSize.y);
```
Bug? vflip is never used. Also naming...

egomeh replied:
Oh that was a mistake, forgot to remove it.

gokselgoktas commented on Assets/Scripts/RayMarching.cs:
```diff
@@ -109,6 +109,7 @@ void OnPreRender()
         {
             m_CommandBuffer = new CommandBuffer();
             m_CommandBuffer.name = "Ray Marching";
+            m_CommandBuffer.SetGlobalFloat("_YAxisRaySign", SystemInfo.graphicsDeviceVersion.StartsWith("Direct") ? -1f : 1f);
```
No better way of doing this? I won't allow this, shaders have macros defined that identify the platform

egomeh replied:
There was, now it's checked via the defines.

egomeh committed:
Changed chack for Direct3D as define in shader
The check for Direct3D is now done as an if condition on the macro defining the shader API
